src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnCache.java

```diff
@@ -292,7 +292,7 @@ private void report(Event evt) {
   }
 
   private static boolean useRemoteCache(RemoteOptions options) {
-    return !isNullOrEmpty(options.remoteCache);
+    return !isNullOrEmpty(options.remoteCache) || !isNullOrEmpty(options.remoteExecutor);
```

Comment:
This also feels slightly wrong, perhaps the function needs to be a class method to allow the `RemoteExecutionCache` to override it.

Reply:
The class structure here is a bit more complex than I'd like, but this seems fine for now.

Reply:
Oh, I have another concern - the remote executor may not allow writes to the action cache. In that case, we can't use remote caching even if it's enabled here.

Reply:
In that case, would it not be a requirement for the bazel client to be ran with [--noremote_upload_local_results](https://docs.bazel.build/versions/master/command-line-reference.html#flag--remote_upload_local_results)?

In our case, we do not currently place this restriction on our remote executor frontend as we're currently transitioning to remote execution and still have some rules which don't work remotely.

I just tested this use case, if bazel attempts to push to the remote executor which rejects action cache writes, the output is as seen below (with the second line being a response from the remote executor.

```
WARNING: Writing to Remote Cache:
UNIMPLEMENTED: This service can only be used to get action results for instance "remote-execution"
```

Reply:
IIUC, this change implicitly enables remote caching if a remote executor is enabled, so people who are relying on the current behavior will start to see those warnings, and they'll see one warning for every such action.

I suppose asking them to explicitly also set ```--noremote_upload_local_results``` is a reasonable workaround, as long as Bazel doesn't crash (as you said, and which I also double-checked in the source code).

Reply:
The change will also effect those who are relying on the current behaviour of not remotely caching `no-remote-exec` actions (To which I'd suggest using `no-remote` as the docs already detail).